src/librustc/ty/query/config.rs

```diff
@@ -13,19 +15,15 @@ use std::borrow::Cow;
 use std::fmt::Debug;
 use std::hash::Hash;
 
-// Query configuration and description traits.
-
-// FIXME(eddyb) false positive, the lifetime parameter is used for `Key`/`Value`.
-#[allow(unused_lifetimes)]
-pub trait QueryConfig<'tcx> {
+pub trait QueryConfig<CTX> {
```

Comment:
Does plain `trait QueryConfig` work here?

Reply:
No, the `Key` and `Value` types may need to access the `'tcx` lifetime from `CTX = TyCtxt<'tcx>`.
I should put a comment back.